task1.c

```diff
@@ -0,0 +1,76 @@
+#include <stdio.h>
+#include <stdbool.h>
+#define BUFFER_SIZE 1000
+
+int main()
+{
+    printf("This program counts total cost of order in file order.txt \n");
+    printf("(file should not contain empty lines, there should be no spaces in price and amount of product) \n");
+    FILE* file = fopen("order.txt", "r");
+    if (file == NULL)
+    {
+        printf("Error opening order.txt.");
+        return -1;
+    }
+    int totalCost = 0;
+    char curString[BUFFER_SIZE] = {0};
+    bool isStringRead = fgets(curString, BUFFER_SIZE, file);
+    int stringNumber = 1;
+    while (isStringRead)
+    {
+        int i = 0;
+        int j = 0;
+        int k = 0;
```

Comment:
```suggestion
        int nameLength = 0;
        int amountLength = 0;
        int priceLength = 0;
```
Если переменные используются сложнее, чем просто для итерации по массиву, у них должны быть содержательные имена. А если считывать посимвольно из файла, то эти переменные вообще не понадобятся.